lib/SILGen/SILGenPattern.cpp

```diff
@@ -1031,18 +1031,25 @@ void PatternMatchEmission::emitWildcardDispatch(ClauseMatrix &clauses,
   bool hasGuard = guardExpr != nullptr;
   assert(!hasGuard || !clauses[row].isIrrefutable());
 
-  auto stmt = clauses[row].getClientData<CaseStmt>();
-  ArrayRef<CaseLabelItem> labelItems = stmt->getCaseLabelItems();
-  bool hasMultipleItems = labelItems.size() > 1;
-  
+  auto stmt = clauses[row].getClientData<Stmt>();
+  bool hasMultipleItems = false;
+  if (isa<CaseStmt>(stmt)) {
+    hasMultipleItems = dyn_cast<CaseStmt>(stmt)->getCaseLabelItems().size() > 1;
+  }
+
   // Bind the rest of the patterns.
   bindIrrefutablePatterns(clauses[row], args, !hasGuard, hasMultipleItems);
 
   // Emit the guard branch, if it exists.
   if (guardExpr) {
-    SGF.usingImplicitVariablesForPattern(clauses[row].getCasePattern(), stmt, [&]{
-      this->emitGuardBranch(guardExpr, guardExpr, failure);
-    });
+    if (isa<CaseStmt>(stmt)) {
+      SGF.usingImplicitVariablesForPattern(clauses[row].getCasePattern(), dyn_cast<CaseStmt>(stmt), [&]{
+        this->emitGuardBranch(guardExpr, guardExpr, failure);
+      });
+    } else {
+      assert(isa<CatchStmt>(stmt));
+      emitGuardBranch(guardExpr, guardExpr, failure);
+    }
```

Comment:
Why is this branch necessary? It seems cleaner to make `usingImplicitVariablesForPattern` do the right thing if the `stmt` is null (as it would be if the `dyn_cast` fails).